src/Google.Datastore.V1Beta3/ArrayValuePartial.cs

```diff
@@ -0,0 +1,346 @@
+        public static implicit operator ArrayValue(DateTimeOffset[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Each value within <paramref name="values"/> is converted
+        /// to a <see cref="Value"/> using the individual implicit conversion. Any null input values are converted
+        /// to instances of <see cref="Value"/> with a kind of Null.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(DateTimeOffset?[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Any null input values are converted
+        /// to instances of <see cref="Value"/> with a kind of Null.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(Value[] values) => ToArrayValue(values, x => x);
+
+        private static ArrayValue ToArrayValue<T>(T[] values, Func<T, Value> converter) =>
+            values == null ? null : new ArrayValue { Values = { values.Select(x => converter(x) ?? Value.ForNull()) } };
+
+        /// <summary>
+        /// Converts an <see cref="ArrayValue"/> to an array. Each value within <paramref name="arrayValue"/> is converted
+        /// to a <see cref="Value"/> using the individual explicit conversion. Any <see cref="Value"/> elements with a kind of
+        /// Null are converted to null values in the resulting array.
+        /// </summary>
+        /// <param name="arrayValue">The array value to convert.</param>
+        /// <returns>An array with converted values, or <c>null</c> if <paramref name="arrayValue"/> is null.</returns>
+        /// <exception cref="InvalidOperationException">An element in the array value does not have the expected kind.</exception>
+        public static explicit operator string[](ArrayValue arrayValue) => FromArrayValue(arrayValue, x => (string) x);
+        /// <summary>
+        /// Converts an <see cref="ArrayValue"/> to an array. Each value within <paramref name="arrayValue"/> is converted
+        /// to a <see cref="Value"/> using the individual explicit conversion.
+        /// </summary>
+        /// <param name="arrayValue">The array value to convert.</param>
+        /// <returns>An array with converted values, or <c>null</c> if <paramref name="arrayValue"/> is null.</returns>
+        /// <exception cref="InvalidOperationException">An element in the array value does not have the expected kind.</exception>
```

Comment:
Looks like this (and all other conversions to arrays of value types) can also throw ArgumentNullException (in Value.CheckKindForNonNullable). Should that be listed here as well or do we assume there won't be nulls based on what you just said about it never being valid?